### PR TITLE
Fixed issue with account metadata copy

### PR DESCRIPTION
**Pull request by rohitnarula7176**
*Added a deep copy extension on nsdictionary.

### CODE REVIEW

File: IdentityCore/src/util/NSDictionary+MSIDExtensions.m [hunk 1]
```diff
@@ -239,4 +239,25 @@ - (id)msidObjectForKey:(NSString *)key ofClass:(Class)requiredClass
...
+- (NSMutableDictionary *) mutableDeepCopy
+{
+    NSMutableDictionary *returnDict = [[NSMutableDictionary alloc] initWithCapacity:[self count]];
+    NSArray *keys = [self allKeys];
+    for (id key in keys)
+    {
+        id value = [self valueForKey:key];
+        id copy = nil;
+        if ([value respondsToSelector:@selector(mutableDeepCopy)])
+            copy = [value mutableDeepCopy];
```
Comment (jasoncoolmax) [inline IdentityCore/src/util/NSDictionary+MSIDExtensions.m]: nit: add curly bracket for the `if` :)

File: IdentityCore/src/util/NSDictionary+MSIDExtensions.m [hunk 2]
```diff
@@ -239,4 +239,25 @@ - (id)msidObjectForKey:(NSString *)key ofClass:(Class)requiredClass
...
+        else if ([value respondsToSelector:@selector(mutableCopy)])
+            copy = [value mutableCopy];
+        if (copy == nil)
+            copy = [value copy];
+        
+        [returnDict setObject:copy forKey:key];
```
Comment (jasoncoolmax) [inline IdentityCore/src/util/NSDictionary+MSIDExtensions.m]: better to do nil check before we call `setObject`? :)

File: IdentityCore/src/util/NSDictionary+MSIDExtensions.m [hunk 3]
```diff
@@ -239,4 +239,25 @@ - (id)msidObjectForKey:(NSString *)key ofClass:(Class)requiredClass
     return nil;
 }
 
+- (NSMutableDictionary *) mutableDeepCopy
```
Comment (oldalton) [inline IdentityCore/src/util/NSDictionary+MSIDExtensions.m]: nit: remove space before "mutableDeepCopy"

Comment (rohitnarula7176) [reply IdentityCore/src/util/NSDictionary+MSIDExtensions.m]: Updated.

File: IdentityCore/src/util/NSDictionary+MSIDExtensions.m (hunk 2 above)
Comment (rohitnarula7176) [reply IdentityCore/src/util/NSDictionary+MSIDExtensions.m]: I think this should be fine as the copy should return a non nil object.

File: IdentityCore/src/util/NSDictionary+MSIDExtensions.m (hunk 1 above)
Comment (rohitnarula7176) [reply IdentityCore/src/util/NSDictionary+MSIDExtensions.m]: Updated.

**Pull request merged**